designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt

```diff
@@ -277,24 +280,49 @@ internal fun resolveVariantsRecursively(
     // zip through all of them after layout and render them in the right location.
     val replacementContent = customizations.getContent(view.name)
     val replacementComponent = customizations.getComponent(view.name)
-    if (replacementContent != null || replacementComponent != null) {
+    if (replacementComponent != null) {
         composableList.add(
             SquooshChildComposable(
                 component = replacementComponent,
-                content = replacementContent,
                 node = resolvedView,
                 parentComponents = parentComps
             )
         )
         // Make sure that the renderer knows that it needs to do an external render for this
         // node.
         resolvedView.needsChildRender = true
+    } else if (replacementContent != null) {
+        // Replacement Content represents a (short, non-virtualized) list of child composables.
+        // We want these child composables to be laid out inside of this container using the
+        // layout properties set up in the DesignCompose document (so if a designer set the list
+        // to wrap and center align its children, then the Composables coming in here should be
+        // presented in that way.
+        //
+        // To do this, we must synthesize a ResolvedSquooshNode for each child, and also consult
+        // its intrinsic size before we perform layout.
+        var previousReplacementChild: SquooshResolvedNode? = null
+        for (idx in 0 ..< replacementContent.count) {
+            val childComponent = replacementContent.content(idx)
+            val replacementChild =
+                generateReplacementListChildNode(resolvedView, idx, rootLayoutId, layoutIdAllocator)
+            if (previousReplacementChild != null)
+                previousReplacementChild.nextSibling = replacementChild
+            else resolvedView.firstChild = replacementChild
+            previousReplacementChild = replacementChild
+
+            composableList.add(
+                SquooshChildComposable(
+                    component = @Composable { childComponent() },
```

Comment:
This is a bit clunky; I'm not sure if there's a nicer way to do this than adding an empty wrapper function, though.